src/coreComponents/physicsSolvers/fluidFlow/StatOutputController.cpp

```diff
@@ -0,0 +1,170 @@
+  DomainPartition & domain = problemManager.getGroup< DomainPartition >( "domain" );
+  TasksManager & taskManager = this->getGroupByPath< TasksManager >( "/Tasks" );
+  OutputManager & outputManager = this->getGroupByPath< OutputManager >( "/Outputs" );
+
+  Group & meshBodies = domain.getMeshBodies();
+  std::vector< string > const groupNames = this->getSubGroupsNames();
+
+  GEOS_ERROR_IF( groupNames.empty() || groupNames.size() >= 2,
+                 "StatOutputController must have one of the following components : {SinglePhaseStatistics, CompositionalMultiphaseStatistics} " );
+
+  m_statistics = &this->getGroup< TaskBase >( groupNames[0] );
+
+  forSubStats( [&]( auto & statistics ) {
+    using STATSTYPE = typename TYPEOFREF( statistics );
+
+    statistics.getSolver()->forDiscretizationOnMeshTargets( meshBodies,
+                                                            [&] ( string const &,
+                                                                  MeshLevel & mesh,
+                                                                  arrayView1d< string const > const & regionNames )
+    {
+      ElementRegionManager & elemManager = mesh.getElemManager();
+      for( string const & regionName : regionNames )
+      {
+        ElementRegionBase & region = elemManager.getRegion( regionName );
+        string const regionStatPath = GEOS_FMT( "{}/regionStatistics", region.getPath() );
+        typename STATSTYPE::RegionStatistics & regionStats = this->getGroupByPath< typename STATSTYPE::RegionStatistics >( regionStatPath );
+
+        string_array packCollectionPaths;
+        regionStats.forWrappers( [&]( WrapperBase const & wrapper )
+        {
+          generatePackCollection( taskManager,
+                                  regionName,
+                                  regionStatPath,
+                                  wrapper.getName(),
+                                  packCollectionPaths );
+        } );
+
+        generateTimeHistory( outputManager, regionName, packCollectionPaths );
+      }
+    } );
+  } );
+
+}
+
+Group * StatOutputController::createChild( string const & childKey, string const & childName )
+{
+  GEOS_LOG_RANK_0( "Adding Statistics: " << childKey << ", " << childName );
+  std::unique_ptr< TaskBase > task = TaskBase::CatalogInterface::factory( childKey, childName, this );
+  return &this->registerGroup< TaskBase >( childName, std::move( task ) );
+}
+
+template< typename LAMBDA >
+void StatOutputController::forSubStats( LAMBDA lambda )
+{
+  forSubGroups< SinglePhaseStatistics,
+                CompositionalMultiphaseStatistics >( lambda );
+}
+
+void StatOutputController::expandObjectCatalogs()
+{
+  createChild( SinglePhaseStatistics::catalogName(),
+               SinglePhaseStatistics::catalogName() );
+  createChild( CompositionalMultiphaseStatistics::catalogName(),
+               CompositionalMultiphaseStatistics::catalogName() );
+}
```

Comment:
Here you need to have something like:
```suggestion
static std::set< string > const allowedChildTypes = { SinglePhaseStatistics::catalogName(),
                                                      CompositionalMultiphaseStatistics::catalogName() };

Group * StatOutputController::createChild( string const & childKey, string const & childName )
{
  GEOS_ERROR_IF( allowedChildTypes.count(childKey)!=0,
                 GEOS_FMT("Wrong children \"{}\" in {}, allowed child types are {}",
                 childKey,getDataContext(),stringutilities::join(allowedChildTypes)) );
  GEOS_LOG_RANK_0( "Adding Statistics: " << childKey << ", " << childName );
  std::unique_ptr< TaskBase > task = TaskBase::CatalogInterface::factory( childKey, childName, this );
  return &this->registerGroup< TaskBase >( childName, std::move( task ) );
}

template< typename LAMBDA >
void StatOutputController::forSubStats( LAMBDA lambda )
{
  forSubGroups< SinglePhaseStatistics,
                CompositionalMultiphaseStatistics >( lambda );
}

void StatOutputController::expandObjectCatalogs()
{
  foreach(auto const & key : allowedChildTypes) {
    createChild( key,key );
  }
}
```